lib/lib-table-query/src/main/java/org/sagebionetworks/table/query/util/TableSqlProcessor.java

```diff
@@ -180,5 +188,129 @@ private static OrderingSpecification getCurrentDirection(OrderByClause obc,
+		//add the new where clause to the sql
+		if(newSearchConditionBuilder.length() > 0){
+			SearchCondition facetSearchCondition = SqlElementUntils.createSearchCondition(newSearchConditionBuilder.toString());
+			tableExpression.replaceWhere(new WhereClause(facetSearchCondition));
+		}
+		return model.toSql();
+	}
+	
+	/**
+	 * Creates the search condition for a FacetColumnRequest
+	 * @param facetColumnRequest
+	 * @return the search condition string
+	 */
+	public static String createFacetSearchConditionString(FacetColumnRequest facetColumnRequest){
+		if (facetColumnRequest == null){
+			return null;
+		}
+		
+		if (facetColumnRequest instanceof FacetColumnValuesRequest){
+			return createEnumerationSearchCondition((FacetColumnValuesRequest) facetColumnRequest);
+		}else if (facetColumnRequest instanceof FacetColumnRangeRequest){
+			return createRangeSearchCondition((FacetColumnRangeRequest) facetColumnRequest);
+		}else{
+			throw new IllegalArgumentException("Unexpected instance of FacetColumnRequest");
+		}
+		
+	}
+	
+	private static String createRangeSearchCondition(FacetColumnRangeRequest facetRange){
+		if(facetRange == null || ( (facetRange.getMin() == null || facetRange.getMin().equals(""))
+									&& (facetRange.getMax() == null || facetRange.getMax().equals("")) ) ){
+			return null;
+		}
+		String min = facetRange.getMin();
+		String max = facetRange.getMax();
+		
+		StringBuilder builder = new StringBuilder("(");
+		
+		//at this point we know at least one value is not null and is not empty string
+		builder.append(facetRange.getColumnName());
+		if(min == null){ //only max exists
+			builder.append("<=");
+			appendValueToStringBuilder(builder, max);
+		}else if (max == null){ //only min exists
+			builder.append(">=");
+			appendValueToStringBuilder(builder, min);
+		}else{
+			builder.append(" BETWEEN ");
+			appendValueToStringBuilder(builder, min);
+			builder.append(" AND ");
+			appendValueToStringBuilder(builder, max);
+		}
+		
+		builder.append(")");
+		return builder.toString();
+	}
+	
+	private static String createEnumerationSearchCondition(FacetColumnValuesRequest facetValues){
+		if(facetValues == null || facetValues.getFacetValues() == null|| facetValues.getFacetValues().isEmpty()){
+			return null;
+		}
+		
+		StringBuilder builder = new StringBuilder("(");
+		int initialSize = builder.length();
+		for(String value : facetValues.getFacetValues()){
+			if(builder.length() > initialSize){
+				builder.append(" OR ");
+			}
+			builder.append(facetValues.getColumnName());
+			if(value.equals(NULL_VALUE_KEYWORD)){
+				builder.append(" IS NULL");
+			}else{
+				builder.append("=");
+				appendValueToStringBuilder(builder, value);
+			}
+		}
+		builder.append(")");
+		return builder.toString();
+	}
+	
+	/**
+	 * Appends a value to the string builder
+	 * and places single quotes (') around it if the string contains spaces
+	 */ 
+	private static void appendValueToStringBuilder(StringBuilder builder, String value){

```

Comment:
This will not enough.  Spaces are just one of many characters that would required quotes. If the ColumnType is STRING then the value needs to be in quotes.